### PR TITLE
FIX #2587 : added more style support in container property pane

### LINKED ISSUE #2587: Container styles
Containers should have more styles than the current rounded borders and shadow

### PR DESCRIPTION
**Pull request by vibhandikyash**
- Outline colour / Border colour: Should have the colour picker
- Outline stoke / Border width: The width of border used
- Corner radius / Border radius: The radius/curve on each corner of the container
- Fill Colour / Background colour: Should have the colour picker

Fixes #2587 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: FIX/2587-container-styles 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.94 **(-0.01)** | 33.75 **(-0.02)** | 30.02 **(0)** | 52.59 **(-0.01)**
 :red_circle: | app/client/src/components/designSystems/appsmith/ContainerComponent.tsx | 93.55 **(0.22)** | 56.52 **(-2.57)** | 100 **(0)** | 93.55 **(0.22)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.59 **(-0.24)** | 39.41 **(-0.84)** | 36.21 **(0)** | 54.77 **(-0.27)**
 :red_circle: | app/client/src/widgets/FormWidget.tsx | 31.25 **(-4.04)** | 0 **(0)** | 8.33 **(-7.05)** | 31.82 **(-4.35)**</details>

### CODE REVIEW

**Review by somangshu — CHANGES_REQUESTED** [review]
Hey I think we should be using developer savvy terminologies like:
- Outline colour -> Border colour
- Outline stoke -> Border width
- Corner radius -> Border radius
- Fill Colour -> Background colour

@riodeuno thoughts?

File: app/client/src/widgets/ContainerWidget.tsx [hunk 1]
```diff
@@ -56,6 +46,46 @@ class ContainerWidget extends BaseWidget<
...
+            placeholderText: "#FFFFFF / Gray / rgb(255, 99, 71)",
+            propertyName: "borderColor",
+            label: "Outline Color",
+            controlType: "COLOR_PICKER",
+            isBindProperty: true,
+            isTriggerProperty: false,
+            validation: VALIDATION_TYPES.TEXT,
+          },
+          {
+            helpText: "enter value for outline strock",
```
Comment (riodeuno) [inline app/client/src/widgets/ContainerWidget.tsx:60]: Let's fix the spelling and the casing here.

File: app/client/src/widgets/ContainerWidget.tsx [hunk 2]
```diff
@@ -56,6 +46,46 @@ class ContainerWidget extends BaseWidget<
...
+            propertyName: "borderWidth",
+            label: "Outline Strock",
```
Comment (riodeuno) [inline app/client/src/widgets/ContainerWidget.tsx:62]: Same typo here.

File: app/client/src/widgets/ContainerWidget.tsx [hunk 3]
```diff
@@ -56,6 +46,46 @@ class ContainerWidget extends BaseWidget<
...
+            placeholderText: "Enter value",
+            controlType: "INPUT_TEXT",
+            isBindProperty: true,
+            isTriggerProperty: false,
+            validation: VALIDATION_TYPES.TEXT,
```
Comment (riodeuno) [inline app/client/src/widgets/ContainerWidget.tsx:67]: @somangshu 
Are we expecting users to enter the `px` suffix as well? If that's the case users can potentially use `cm` as a suffix as well. Do we want this? Also what happens when users don't add a suffix at all and just a number? Can we detail these in the `helpText` and `placeholderText` fields of this property?

File: app/client/src/widgets/ContainerWidget.tsx [hunk 4]
```diff
@@ -56,6 +46,46 @@ class ContainerWidget extends BaseWidget<
...
+          },
+          {
+            helpText: "enter value for corner radius",
+            propertyName: "borderRadius",
+            label: "Corner Radius",
+            placeholderText: "Enter value",
```
Comment (riodeuno) [inline app/client/src/widgets/ContainerWidget.tsx:73]: The placeholder needs to be quite specific here. Users can possibly use `%`, `px`, `cm`, etc

**Review by riodeuno — CHANGES_REQUESTED** [review]

File: app/client/src/widgets/ContainerWidget.tsx (hunk 3 above)
Comment (somangshu) [reply app/client/src/widgets/ContainerWidget.tsx:67]: @riodeuno now I feel we should do something like the list widget, and fix the unit to `px` and let only integer be allowed

**Review by somangshu — DISMISSED** [review]
LGTM

**Comment by somangshu** [comment]
@vibhandikyash related cypress tests are failing here, Can you check?

**Review by somangshu — DISMISSED** [review]

**Comment by somangshu** [comment]
@riodeuno is the failing test case related to some drag & drop change

**Comment by somangshu** [comment]
@vibhandikyash we need to have a discussion here to understand if this can be taken forward now. @marks0351 will be able to provide us some clarity

**Comment by marks0351** [comment]
@vibhandikyash just mentioning that we had discussed this.
cc: @somangshu

**Comment by vibhandikyash** [comment]
Will close this and continue it futher at #6476

**Pull request closed without merging**